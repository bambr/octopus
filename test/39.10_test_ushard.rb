#!/usr/bin/ruby
# coding: utf-8

$: << File.dirname($0)
require '39_test_ushard'

$one.create_shard 1, :POR, "one", "three"
$one.create_shard 2, :POR, "one", "two"
$two.create_shard 3, :POR, "two", "three"

$one.create_object_space 0, :shard => 1, :index => DEFIDX
$one.insert [1,"one"], :shard => 1

$one.create_object_space 0, :shard => 2, :index => DEFIDX
$one.insert [1,"one"], :shard => 2

$one.create_object_space 0, :shard => 3, :index => DEFIDX
$one.insert [1,"one"], :shard => 3

$three = ThreeEnv.new.connect_eval do
  ping
  self
end

sleep 0.1

$three.select 1, :shard => 1
$three.select 1, :shard => 3
