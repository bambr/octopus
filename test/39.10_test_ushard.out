# one.ping()
:pong

# two.ping()
:pong

# one.create_shard(1, :POR, "one", "three")
:success

# one.create_shard(2, :POR, "one", "two")
:success

# two.create_shard(3, :POR, "two", "three")
:success

# one.create_object_space(0, {:shard=>1, :index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
:success

# one.insert([1, "one"], {:shard=>1})
1

# one.create_object_space(0, {:shard=>2, :index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
:success

# one.insert([1, "one"], {:shard=>2})
1

# one.create_object_space(0, {:shard=>3, :index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
:success

# one.insert([1, "one"], {:shard=>3})
1

# box.ping()
:pong

# box.select(1, {:shard=>1})
[["\x01\x00\x00\x00", "one"]]

# box.select(1, {:shard=>3})
[["\x01\x00\x00\x00", "one"]]

